examples/porting/lv_port_disp_template.c

```diff
@@ -63,22 +63,25 @@ void lv_port_disp_init(void)
 
     /* Example 1
      * One buffer for partial rendering*/
-    static lv_color_t buf_1_1[MY_DISP_HOR_RES * 10];                          /*A buffer for 10 rows*/
-    lv_display_set_draw_buffers(disp, buf_1_1, NULL, sizeof(buf_1_1), LV_DISPLAY_RENDER_MODE_PARTIAL);
+    LV_DRAW_BUF_DEFINE(buf_1_1, MY_DISP_HOR_RES, MY_DISP_VER_RES / 10, LV_COLOR_FORMAT_NATIVE); /*A buffer for 1/10 fb*/
```

Comment:
What about using `lv_draw_buf_init()`? It's more explicit about what is happening and provides more control on the buffer allocation/settings. We can use `_LV_DRAW_BUF_SIZE` to calculate the size of the buffers.

If this macro will be used here the `_` should be removed too.

---

Or we can show both options. ` LV_DRAW_BUF_DEFINE` as a simple one, and in the next mode `lv_draw_buf_init` as a more advanced option.

Reply:
I agree. The `LV_DRAW_BUF_DEFINE ` is actually not working on platforms with GPU. We tried to fix today but found no solution to align in preprocessing stage. 

Either we can introduce compiler flags, or we call `lv_draw_buf_init()` instead.

I have modified `LV_DRAW_BUF_DEFINE ` locally to include call to `lv_draw_buf_init` and it works. Do you guys think it's acceptable?


```c
/**
 * Define a static draw buffer with the given width, height, and color format.
 * Stride alignment is set to LV_DRAW_BUF_STRIDE_ALIGN.
 */
#define LV_DRAW_BUF_DEFINE(name, _w, _h, _cf) \
    static uint8_t buf_##name[_LV_DRAW_BUF_SIZE(_w, _h, _cf)]; \
    static lv_draw_buf_t name;
    lv_draw_buf_init(&name, _w, _h, _cf, 0, buf_##name, sizeof(buf_##name));

```

Reply:
> What about using lv_draw_buf_init()? It's more explicit about what is happening and provides more control on the buffer allocation/settings. 

OK, no problem.

Reply:
> I have modified `LV_DRAW_BUF_DEFINE` locally to include call to lv_draw_buf_init and it works. Do you guys think it's acceptable?

If people use this macro outside a function, it would be a problem.

Reply:
It confirms my concern raised [here](https://github.com/lvgl/lvgl/pull/5204#issuecomment-1884744920).

Reply:
Updated. Please review.